### DISCUSSION TITLE
Tesla API Signup for Euro/UK requires TAX ID

### DISCUSSION THREAD

**Discussion by gareth78** [llamafilm/tesla-http-proxy-addon#34] [General]
It seems it's not possible to signup for the devloper access in the UK (and a few EU countries i tried) as the signup process has a mandatory requirement for a TAX ID

![image](https://github.com/llamafilm/tesla-http-proxy-addon/assets/52722211/54074130-4058-4a50-bcbb-84caeb268e30)

**Comment by iainbullock** [comment]
Just entering GB00000000 or GB0000000000 or similar worked for me

**Reply by noobey** [reply]
Same issue.
Weird thing is, I have an actual company with a Company Number and VAT number.
Neither worked. Neither did GB00000000 or GB0000000000.

**Comment by llamafilm** [comment]
Moving to discussion because this is not a problem with the addon.

In the US, every citizen has a social security number which can be used as tax ID. Maybe you have something similar, or if not, I suppose there must be a way for you obtain a tax ID.

**Reply by anandlar** [reply]
In Sweden you have to registrate a company to get a tax id. It comes with a lot of administration. We have a personal social number (similar to tax id), but it doesn't work. Any suggestions?

**Comment by dominikandreas** [comment]
```python
from random import randint
 
random_numbers = tuple(randint(0, 9) for _ in range(8))

# Calculate the check digit (sum of numbers % 10)
check_digit = sum(random_numbers) % 10

# Form the full VAT ID
print("DE" + ''.join(map(str, random_numbers)) + str(check_digit))
```

This python snippet worked for me. You can run it here to generate a VAT ID: https://www.online-python.com/0nqvfUlRDM

**Reply by anandlar** [reply]
Thanks, but sadly it doesnt work for me. Any tips?

**Reply by dominikandreas** [reply]
This is for German Tax IDs, you'd have to adapt it to the format used on Sweden. Maybe the following works for you?

```python
from random import randint
 
random_numbers = tuple(randint(0, 9) for _ in range(9))

# Calculate the check digit (sum of numbers % 10)
check_digit = sum(random_numbers) % 10

# Form the full VAT ID
print("SE" + ''.join(map(str, random_numbers)) + str(check_digit) +"01")
```
https://www.online-python.com/d2QVxk5Ony

Links to information I've used for this:
https://en.m.wikipedia.org/wiki/VAT_identification_number
https://sv.m.wikipedia.org/wiki/Momsregistreringsnummer
https://sv.m.wikipedia.org/w/index.php?title=Organisationsnummer&wprov=rarw1

Edit: according to some.online vat checkers, the generated IDs are not correct, maybe the checksum needs to be calculated differently. Couldn't find any information about this, but maybe you can figure it out
